### PR TITLE
[Merged by Bors] - feat: add missing WithTop/WithBot lemmas and improve WithBot Nat proofs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Monoid/WithTop.lean [hunk 1]
```diff
@@ -373,6 +373,17 @@ instance addMonoidWithOne : AddMonoidWithOne (WithTop α) :=
 @[deprecated (since := "2024-04-05")] alias nat_ne_top := natCast_ne_top
 @[deprecated (since := "2024-04-05")] alias top_ne_nat := top_ne_natCast
 
+-- See note [no_index around OfNat.ofNat]
+@[simp] lemma coe_ofNat (n : ℕ) [n.AtLeastTwo] :
+    (no_index (OfNat.ofNat n : α) : WithTop α) = OfNat.ofNat n := rfl
+@[simp] lemma coe_eq_ofNat (n : ℕ) [n.AtLeastTwo] (m : α) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Monoid/WithTop.lean:379]: I guess we should have `ofNat_eq_coe` too? (and same for WithBot`)

**Review by eric-wieser — APPROVED** [review]
Are there any inequality lemmas about `ofNat` you can add at the same time?

bors d+

**Comment by Ruben-VandeVelde** [comment]
Possibly there's more we could add, but I'll stick with these at the moment.

bors merge

File: Mathlib/Algebra/Order/Monoid/WithTop.lean [hunk 2]
```diff
@@ -373,6 +373,20 @@ instance addMonoidWithOne : AddMonoidWithOne (WithTop α) :=
 @[deprecated (since := "2024-04-05")] alias nat_ne_top := natCast_ne_top
 @[deprecated (since := "2024-04-05")] alias top_ne_nat := top_ne_natCast
 
+-- See note [no_index around OfNat.ofNat]
+@[simp] lemma coe_ofNat (n : ℕ) [n.AtLeastTwo] :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Monoid/WithTop.lean:377]: I think this and some others should be `norm_cast`

**Pull request closed without merging**